go-controller/pkg/cni/cniserver.go

```diff
@@ -222,13 +230,16 @@ func (s *Server) handleCNIRequest(r *http.Request) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
+	if s.mode == types.NodeModeSmartNICHost {
```

Comment:
I wanted to avoid reaching into the globally defined structs in config package if possible as imo its a bit clearer if cniserver bases its decisions on the values of internal members instead of globally defined values.

Reply:
OK up to you, but in my experience we typically allow config var direct access across the project